src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -150,6 +213,237 @@ def get_S3FileSystem_from_session(
     return s3_fs
 
 
+def get_export_end_date(filename: str) -> Union[str, None]:
+    """Gets the export end date (converted to isoformat) based on the filename.
+
+        We can have filenames with with:
+        - {DateType}_{YYYYMMDD}-{YYYYMMDD}.json
+        - {DateType}_{YYYYMMDD}.json format
+        This function does handling for both. The export end date is always the
+        last component of the filename.
+
+    Args:
+        filename (str): name of the input json file
+
+    Returns:
+        Union[str, None]: export end date in isoformat if it exists
+    """
+    if (
+        filename is None
+        or filename == ""
+        or len(os.path.splitext(filename)[0].split("_")) <= 1
+    ):
+        return None
+
+    filename_components = os.path.splitext(filename)[0].split("_")
+
+    if "-" in filename_components[-1]:
+        _, end_date = filename_components[-1].split("-")
+        end_date = datetime.datetime.strptime(end_date, "%Y%m%d")
+    else:
+        end_date = datetime.datetime.strptime(filename_components[-1], "%Y%m%d")
+    export_end_date = end_date.isoformat()
+    return export_end_date
+
+
+def get_cohort_from_s3_uri(s3_uri: str) -> Union[str, None]:
+    """Gets the cohort (pediatric_v1 or adults_v1)
+    from the s3 uri of the export.
+
+    The s3 uri of the export has the following expected format:
+    s3://{bucket_name}/{namespace}/{cohort}/{export_name}
+
+    Args:
+        s3_uri (str): the S3 uri
+
+    Returns:
+        Union[str, None]: the cohort if it exists
+    """
+    cohort = None
+    if ADULTS  in s3_uri:
+        cohort = ADULTS
+    elif PEDIATRIC  in s3_uri:
+        cohort = PEDIATRIC
+    else:
+        pass
+    return cohort
+
+
+def get_data_type_from_filename(filename: str) -> Union[str, None]:
+    """Gets the data type from the JSON filename.
+
+    A filename can be of the following format:
+    {DataType}_[{DataSubType}_][Deleted_]{YYYYMMDD}[-{YYYYMMDD}].json
+
+    Since we don't have support for DataSubType, we only support top
+    level datatypes but because we only have the subtypes
+    of the dataset in the exports, we can still use that to get the metadata
+    we need. We parse out deleted healthkit data types as we don't process them
+    into parquet.
+
+    Args:
+        filename (str): JSON filename
+
+    Returns:
+        Union[str, None]: the data type otherwise returns None as we
+        currently don't support comparingsubtypes in data
+    """
+    filename_components = os.path.splitext(filename)[0].split("_")
+    data_type = filename_components[0]
+
+    if "HealthKitV2" in data_type and filename_components[-2] == "Deleted":
```

Comment:
Are we certain that "Deleted' will always be in that spot of the file?

Reply:
It is our expected filename format according to the data docs: https://sagebionetworks.jira.com/wiki/spaces/DPE/pages/3062398977/Data+Dictionary#Derivation. But if they ever decide to suddenly change the format, then yes it wouldn't meet the if statement criteria anymore. But this is OK for this code because deleted subtypes aren't in our `INDEX_FIELD_MAP` so it would change to returning `None` which is also okay since we won't be comparing deleted data types.

Reply:
@thomasyu888 Yes, at least that's the format CE has used so far (they don't provide an API definition or anything). And there is always an underscore in the filename to separate the data type name from the date, so the `.[-2]` element will always exist (same caveat as previous sentence).